### PR TITLE
Checkstyle plugin

### PR DESCRIPTION
**Pull request by AustinShalit**
I added the checkstyle plugin.  This should help solve #261.

### CODE REVIEW

**Comment by JLLeitschuh** [comment]
Can you make it so that intellij automatically imports the stylesheet when you use `./gradlew idea`
Perhaps @JacisNonsense might know more about that.
Also, make sure that all of the project files pass the style check.

**Comment by JLLeitschuh** [comment]
Also, do we want to make this something to make `./gradlew check` fail? If so then the task `check` needs to depend upon checkstyle.

File: build.gradle [hunk 1]
```diff
@@ -28,9 +29,13 @@ allprojects {
...
         toolVersion = "0.7.5.201505241946"
     }
-
+    checkstyle {
+        configFile = file("${project.rootDir}/google_checks.xml")
+        reportsDir = file("${project.rootDir}/build/reports/checkstyle/")
+    }
 
     dependencies {
+        compile checkstyle('com.puppycrawl.tools:checkstyle:6.14')
```
Comment (JLLeitschuh) [inline build.gradle]: I don't think you want this here. Not entirely sure.

Comment (AustinShalit) [inline build.gradle]: You are right.  I fixed this.

**Comment by ThomasJClark** [comment]
How do we run checkstyle?  I can't find it under `gradle tasks`

**Comment by AustinShalit** [comment]
I have been running it by running: 
```
./gradlew :ui:check
```
This PR is failing right now for some reason.  Will check it out later.

**Comment by AustinShalit** [comment]
We should not merge this until after the V1.0.0 release.

**Comment by JacisNonsense** [comment]
I'll take a look at IntelliJ integration when I get the chance

**Comment by JLLeitschuh** [comment]
I agree, this should wait until after the kickoff release.

File: build.gradle [hunk 2]
```diff
@@ -49,6 +49,12 @@ allprojects {
         }
     }
 
+    checkstyle {
+        configFile = file("${project.rootDir}/google_checks.xml")
+        reportsDir = file("${project.rootDir}/build/reports/checkstyle/")
+        toolVersion = '6.14'
+    }
+
     check.dependsOn jacocoTestReport
```
Comment (JacisNonsense) [inline build.gradle]: Maybe `check.dependsOn build`? This should work unless there is a circular task dependency

Comment (JLLeitschuh) [inline build.gradle]: `check.dependsOn build jacocoTestReport`

**Comment by AustinShalit** [comment]
I am going to close this PR because Codacy now handles style checking.

**Comment by JLLeitschuh** [comment]
I should configure that at some point.

**Comment by AustinShalit** [comment]
I have a configuration that I can send you if you do not want to start with a blank slate.  You can view the issues it generates here: https://www.codacy.com/app/austinshalit/GRIP/dashboard

**Comment by JLLeitschuh** [comment]
Yea, if you could send me the config that would be great.

**Pull request closed without merging**